iocore/net/SSLSessionCache.cc

```diff
@@ -158,12 +158,23 @@ SSLSessionBucket::insertSession(const SSLSessionID &id, SSL_SESSION *sess)
   }
 
   Ptr<IOBufferData> buf;
-  buf = new_IOBufferData(buffer_size_to_index(len, MAX_BUFFER_SIZE_INDEX), MEMALIGNED);
+  Ptr<IOBufferData> buf_exdata;
+  size_t len_exdata = sizeof(ssl_session_cache_exdata);
+  buf               = new_IOBufferData(buffer_size_to_index(len, MAX_BUFFER_SIZE_INDEX), MEMALIGNED);
   ink_release_assert(static_cast<size_t>(buf->block_size()) >= len);
   unsigned char *loc = reinterpret_cast<unsigned char *>(buf->data());
   i2d_SSL_SESSION(sess, &loc);
+  buf_exdata = new_IOBufferData(buffer_size_to_index(len, MAX_BUFFER_SIZE_INDEX), MEMALIGNED);
+  ink_release_assert(static_cast<size_t>(buf_exdata->block_size()) >= len_exdata);
+  ssl_session_cache_exdata *exdata = reinterpret_cast<ssl_session_cache_exdata *>(buf_exdata->data());
+  // This could be moved to a function in charge of populating exdata
+#ifndef OPENSSL_IS_BORINGSSL
+  exdata->curve = SSL_get_shared_curve(ssl, 0);
```

Comment:
nitpick: These functions are called from two places now, so I may want to have a function that absorb the parameter difference between `SSL_get_shared_curve` and `SSL_get_curve_id` to reduce ifdef.

Reply:
done, thanks!